Tests/NIOSSHTests/UserAuthenticationStateMachineTests.swift

```diff
@@ -720,4 +730,27 @@ final class UserAuthenticationStateMachineTests: XCTestCase {
         // Let's say we got a success. Happy path!
         XCTAssertNoThrow(try stateMachine.receiveUserAuthSuccess())
     }
+
+    func testCertificateClientAuthFlow() throws {
+        let delegate = InfinitePrivateKeyDelegate()
+        var stateMachine = UserAuthenticationStateMachine(role: .client(.init(userAuthDelegate: delegate, serverAuthDelegate: AcceptAllHostKeysDelegate())), loop: self.loop, sessionID: self.sessionID)
+
+        XCTAssertNoThrow(try self.beginAuthentication(stateMachine: &stateMachine))
+        stateMachine.sendServiceRequest(.init(service: "ssh-userauth"))
+
+        let dataToSign = UserAuthSignablePayload(sessionIdentifier: self.sessionID, userName: "foo", serviceName: "ssh-connection", publicKey: delegate.key.publicKey)
```

Comment:
This doesn't look like it'll compile. Is there a `key` member on `InfinitePrivateKey` delegate?

Reply:
I'm using the wrong delegate type 🤦🏻‍♂️ give me a minute

Reply:
Fixed!